faunadb/client.py

```diff
@@ -89,6 +119,29 @@ def ping(self, scope=None, timeout=None):
     """
     return self._execute("GET", "ping", query={"scope": scope, "timeout": timeout})
 
+  def new_session_client(self, secret, observer=None):
+    """
+    Create a new client from the existing config with a given secret.
+    The returned client share its parent underlying resources.
+
+    :param secret:
+      Credentials to use when sending requests.
+    :param observer:
+      Callback that will be passed a :any:`RequestResult` after every completed request.
+    :return:
+    """
+    if self.counter.get() > 0:
```

Comment:
You should use `increment` here. There is a chance that `get` returns more than zero but immediately after some other thread decrements and close the session you're sharing with the new client.